src/data-engine.js

```diff
@@ -102,7 +102,7 @@ class DataEngine {
    * @param {HTMLElement} node
    * @param {object} item
    * @param {string} nodeName
-   * @returns {HTMLElement|null}
+   * @returns {Element|null}
```

Comment:
I think the other `HTMLElement`s need changing to `Element` as well

https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

for `createItemElement` etc.

Reply:
I'm afraid I cannot agree with you here, Paul. The other ones are either parameters which should be `HTMLElements`s or are returned values which would again be `HTMLElement`s.

Reply:
good point, I was referring to the MDN docs for `createElement` which said `Element` but I should have clicked through where it then says `HTMLElement`

https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

> Return value
> The new Element.

https://developer.mozilla.org/en-US/docs/Web/API/Element

>  For example, the HTMLElement interface is the base interface for HTML elements,